virtcontainers/bridgedmacvlan_endpoint.go

```diff
@@ -115,3 +116,52 @@ func (endpoint *BridgedMacvlanEndpoint) HotAttach(h hypervisor) error {
 func (endpoint *BridgedMacvlanEndpoint) HotDetach(h hypervisor, netNsCreated bool, netNsPath string) error {
 	return fmt.Errorf("BridgedMacvlanEndpoint does not support Hot detach")
 }
+
+func (endpoint *BridgedMacvlanEndpoint) save() (s persistapi.NetworkEndpoint) {
+	s.Type = string(endpoint.Type())
+	s.BridgedMacvlan = &persistapi.BridgedMacvlanEndpoint{
+		NetPair: persistapi.NetworkInterfacePair{
+			TapInterface: persistapi.TapInterface{
+				ID:   endpoint.NetPair.TapInterface.ID,
+				Name: endpoint.NetPair.TapInterface.Name,
+				TAPIface: persistapi.NetworkInterface{
+					Name:     endpoint.NetPair.TapInterface.TAPIface.Name,
```

Comment:
It would be clearer if you create a variable to refer to `endpoint.NetPair` (and maybe others for referring to `endpoint.NetPair.TapInterface` and `endpoint.NetPair.VirtIface`) I think.

Reply:
The copy is deep so it's hard to be clear, I don't think separate some variables would help 😂

Reply:
I've separated common parts to several functions, and added unit tests.
Please take another look. @jodh-intel